nano-sync/src/primitives/merkle_tree.rs

```diff
@@ -0,0 +1,239 @@
+use std::cmp;
```

Comment:
Zexe has an implementation for merle trees (that also works with Pedersen hashes).
Shouldn't we use that one? Is there anything that prevents us from using their implementation?

Reply:
Ah, was it the issue that theirs starts with a 0 and that is problematic due to the incomplete addition?

Reply:
If so, we could probably just implement a version of Pedersen that works and plug it in into their merkle tree without any issues.

Reply:
To be honest, their version of the Merkle tree is a bit confusing. And ya, we would need to change our Pedersen hash to make it work with Zexe's types and then plug it into their Merkle tree.

Reply:
I just don't see the need. We already have our code, it works and is stable.